Source/ARTConnection.m

```diff
@@ -303,3 +291,49 @@ + (instancetype)newWithConnectionEvent:(ARTRealtimeConnectionEvent)value {
 }
 
 @end
+
+@implementation ARTConnectionRecoveryKey
+
+- ( NSString *)jsonString{
+    NSError *error;
+    NSDictionary *object = @{
+        @"msgSerial": @(_msgSerial),
+        @"connectionKey": _connectionKey,
+        @"serials": _channelSerials ? _channelSerials : @{}
+    };
+    
+    NSData *jsonData = [NSJSONSerialization dataWithJSONObject:object
+                                                       options:0
+                                                         error:&error];
+    if (error) {
+        return nil;
+    } else {
+        return [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+    }
+    return nil;
+}
+
++(ARTConnectionRecoveryKey *)fromJsonString:(NSString *)json{
+        NSData* jsonData = [json dataUsingEncoding:NSUTF8StringEncoding];
+
+        NSError *error = nil;
+        NSDictionary  *object = [NSJSONSerialization
+                                 JSONObjectWithData:jsonData
+                                 options:0
+                                 error:&error];
+
+        if(!error) {
+            ARTConnectionRecoveryKey *recoveryKey = [[ARTConnectionRecoveryKey alloc] init];
```

Comment:
I've commented on the interface to this class, but I feel it should be an immutable instance, so this init should take all properties as args.

Reply:
e062db7a74198d9d0e24b86f2324aa221fd0156e